### PR TITLE
Esplurality

### PR DESCRIPTION
**Pull request by delaray**
Description: Added support for multiple Elasticsearch nodes. Specifically: added smart-code which dispatches to elastisch restful api or native api based on argument type. Added parse-elasticsearch-hosts to parse serialized list of hosts. Added unit tests for each.

Tello card: https://trello.com/c/V0dlGRPh/2542-update-elasticsearch-bolt-to-round-robin-through-nodes

Deployment Instructions: ?

Testing Instructions: lein test 

Reviewers:: @alexkehayias @jeremyheiler

### CODE REVIEW

**Comment by alexkehayias** [comment]
We should just commit to using the native for both situations (pool or single host). It's a little too implicit to decide on the connection type based on the input type. Native drivers should be faster anyway.

File: test/elasticsearch_bolt/storm_test.clj [hunk 1]
```diff
@@ -6,9 +6,30 @@
...
+(def elasticsearch-hosts [["127.0.0.0" 9200]["127.0.0.0" 9200]["127.0.0.0" 9200]])
+
+(deftest test-parse-elasticsearch-hosts
+  (is (second (second (parse-elasticsearch-hosts elasticsearch-hosts-specs))) 9200))
+
+(deftest test-smart-connect
+  ;; Test backward compatibility with restful API.
+  (is (instance? clojurewerkz.elastisch.rest.ElasticSearchEndpoint
+                 (smart-connect elasticsearch-endpoint))
+      true)
```
Comment (jeremyheiler) [inline test/elasticsearch_bolt/storm_test.clj]: `is` expects a boolean value, not a "expected, actual" pair. you'd say `(= a b)` for that.

saying `(is (instance? ...))` is correct

Comment (delaray) [inline test/elasticsearch_bolt/storm_test.clj]: Thanks. Fixed.

**Comment by delaray** [comment]
@alexkehayias Simplified logic to only use native api.

**Comment by jeremyheiler** [comment]
:+1:

File: test/elasticsearch_bolt/storm_test.clj [hunk 2]
```diff
@@ -6,7 +6,20 @@
         [elasticsearch-bolt.fields :only [elasticsearch-input-fields elasticsearch-output-fields]]
         [elasticsearch-bolt.test-data.util :only [with-test-elasticsearch]]
         [elasticsearch-bolt.test-data.parsed.movies :only [movies movie-mappings]])
-  (:require [clojurewerkz.elastisch.query :as q]))
+  (:require [clojurewerkz.elastisch.query :as q]
+            [clojurewerkz.elastisch.rest :as esr]
+            [clojurewerkz.elastisch.native :as esn]))
+
+(def elasticsearch-endpoint "http://127.0.0.1:9200")
+(def elasticsearch-hosts-specs "127.0.0.1:9200;127.0.0.1:9200;127.0.0.1:9200")
```
Comment (alexkehayias) [inline test/elasticsearch_bolt/storm_test.clj]: I was expecting the format to be comma separated not semi-colon

Comment (delaray) [inline test/elasticsearch_bolt/storm_test.clj]: Done.

**Comment by alexkehayias** [comment]
This looks good to go but the pull request needs to be for the dev branch so we can do a snapshot release.

**Pull request closed without merging**